miden-tx/tests/wallet_test.rs

```diff
@@ -66,29 +67,28 @@ fn test_receive_asset_via_wallet() {
     let block_ref = data_store.block_header.block_num();
     let note_ids = data_store.notes.iter().map(|note| note.id()).collect::<Vec<_>>();
 
-    let tx_script_code = ProgramAst::parse(
-        "
-        use.miden::contracts::auth::basic->auth_tx
-
-        begin
-            call.auth_tx::auth_tx_rpo_falcon512
-        end
-        "
-        .to_string()
-        .as_str(),
-    )
-    .unwrap();
+    let tx_script_code = ProgramAst::parse(DEFAULT_AUTH_CODE).unwrap();
     let tx_script = executor
         .compile_tx_script(tx_script_code, vec![(target_pub_key, target_keypair_felt)], vec![])
         .unwrap();
 
     // Execute the transaction and get the witness
-    let transaction_result = executor
+    let executed_transaction = executor
         .execute_transaction(target_account.id(), block_ref, &note_ids, Some(tx_script))
         .unwrap();
 
+    // Prove the transaction
+    let proof_options = ProvingOptions::default();
+    let prover = TransactionProver::new(proof_options);
+    let proven_transaction = prover.prove_transaction(executed_transaction.clone()).unwrap();
+
+    // Verify that the generated proof is valid
+    let verifier = TransactionVerifier::new(MIN_PROOF_SECURITY_LEVEL);
```

Comment:
As a part of these tests, let's also serialize and then deserialize `ProvenTransaction` to simulate it going over the wire.

Reply:
> As a part of these tests, 

Do you mean in these two tests: 

- `test_receive_asset_via_wallet`: 
https://github.com/0xPolygonMiden/miden-base/blob/d885a6e5343fde2aef33623d41ad6aadccce9fc9/miden-tx/tests/wallet_test.rs#L25-L107

- `test_send_asset_via_wallet`:
https://github.com/0xPolygonMiden/miden-base/blob/d885a6e5343fde2aef33623d41ad6aadccce9fc9/miden-tx/tests/wallet_test.rs#L109-L195

Or do you mean in all tests that prove and verify txs?

Reply:
> Or do you mean in all tests that prove and verify txs?

I think whichever tests we end up doing prove-and-very in, we should serialize `ProvenTransaction`'s and then deserialize them before execution (performance overhead of this serialization/deserialization should be negligible).